test/ref.jsx

```diff
@@ -0,0 +1,47 @@
+import { expect } from 'chai';
+import React from 'react';
+
+import { ref } from '..';
+
+import callValidator from './_callValidator';
+
+describe('ref', () => {
+  it('is a function', () => {
+    expect(typeof ref).to.equal('function');
+  });
+
+  function assertPasses(validator, element, propName) {
+    expect(callValidator(validator, element, propName, '"ref" test')).to.equal(null);
+  }
+
+  function assertFails(validator, element, propName) {
+    expect(callValidator(validator, element, propName, '"ref" test')).to.be.instanceOf(Error);
+  }
+
+  it('passes on null/undefined', () => {
+    assertPasses(ref(), <div someRef={undefined} />, 'someRef');
+    assertPasses(ref(), <div someRef={null} />, 'someRef');
+  });
+
+  it('passes with legacy refs', () => {
+    assertPasses(ref(), <div someRef={() => {}} />, 'someRef');
+  });
+
+  it('passes with ref objects', () => {
+    assertPasses(ref(), <div someRef={React.createRef()} />, 'someRef');
+  });
+
+  it('fails with non-refs', () => {
+    assertFails(ref(), <div someRef={666} />, 'someRef');
+  });
+
+  describe('isRequired', () => {
+    it('passes when not required and not present', () => {
+      assertPasses(ref(), <div />, 'someRef');
+    });
```

Comment:
This test feels a bit redundant. Perhaps there's another test organization that would better capture the differences between non-required and required checks. 🤔

Reply:
It does seem redundant with the test on line 21 - i think you can just delete it?

Reply:
Updated tests